docs/sql-reference/sql-statements/generated_column.md

```diff
@@ -0,0 +1,356 @@
+# Generated columns
+
+Since v3.1, StarRocks supports generated columns. Generated columns can be used to accelerate queries with complex expressions. This feature supports precomputing and storing the results of expressions and [query rewrite](#query-rewrite), which significantly accelerates queries with the same complex expressions.
+
+You can define one or more generated columns to store the results of expressions at table creation. As such, when executing queries that contain the expression whose results are stored in the generated column you have defined, the CBO rewrites the query to read data directly from the generated column. Alternatively, you can directly query the data in the generated column.
+
+It is also recommended to **evaluate the impact of generated columns on loading performance because computing expressions takes some time**. Additionally, it is advised to **[create generated columns at table creation](#create-generated-columns-at-table-creation-recommended) rather than adding or modifying them after table creation**. Because it is time-consuming and costly to add or modify generated columns after table creation.
+
+## Basic operations
+
+### Create  generated columns
+
+#### Syntax
+
+```SQL
+<col_name> <data_type> [NULL] AS <expr> [COMMENT 'string']
+```
+
+#### Create generated columns at table creation (Recommended)
+
+Create a table named `test_tbl1` with five columns of which columns `newcol1` and `newcol2` are generated columns whose values are computed by using the specified expressions and referencing the values of the regular columns `data_array` and `data_json` respectively.
+
+```SQL
+CREATE TABLE test_tbl1
+(
+    id INT NOT NULL,
+    data_array ARRAY<int> NOT NULL,
+    data_json JSON NOT NULL,
+    newcol1 DOUBLE AS array_avg(data_array),
+    newcol2 String AS json_string(json_query(data_json, "$.a"))
+)
+Primary KEY (id)
+DISTRIBUTED BY HASH(id);
+```
+
+**NOTICE**:
+
+- Generated columns must be defined after regular columns.
+- Aggregate functions cannot be used in the expressions for generated columns. 
+- The expressions for generated column cannot reference other generated columns or auto-increment columns, but the expressions can reference multiple regular columns.
+- The data type of a generated column must match the data type of the expression's result.
```

Comment:
I suggest changing "the expression's result" to "the result generated by the expression for the generated column", because the logic is easier for users to understand?